ignite/metrics/accuracy.py

```diff
@@ -20,14 +20,8 @@ def reset(self):
 
     def _check_shape(self, output):
         y_pred, y = output
-
-        if y.ndimension() > 1 and y.shape[1] == 1:
-            # (N, 1, ...) -> (N, ...)
-            y = y.squeeze(dim=1)
-
-        if y_pred.ndimension() > 1 and y_pred.shape[1] == 1:
-            # (N, 1, ...) -> (N, ...)
-            y_pred = y_pred.squeeze(dim=1)
+        y_pred = y_pred.squeeze()
+        y = y.squeeze()
```

Comment:
nice :)

Reply:
@vfdev-5 I just realized this fails in case of batch size = 1.

I think we should squeeze all dimensions except for the 1st. Thoughts?

Reply:
make sense, we can try like that

Reply:
@vfdev-5 have a look at the updated code, let me know your thoughts.